packages/adminrouter/extra/src/includes/server/common.conf

```diff
@@ -4,3 +4,20 @@ location /dcos-metadata/dcos-version.json {
     # Allow non-authed access for the UI.
     alias /opt/mesosphere/active/dcos-metadata/etc/dcos-version.json;
 }
+
+## Error pages:
+error_page 401 /error_pages/401.html;
+error_page 403 /error_pages/403.html;
+error_page 404 /error_pages/404.html;
+error_page 500 /error_pages/500.html;
+error_page 501 /error_pages/501.html;
+error_page 502 /error_pages/502.html;
+error_page 503 /error_pages/503.html;
+error_page 504 /error_pages/504.html;
```

Comment:
Do nginx' standard error pages contain nginx version information?

Does this list here replace _all_ nginx standard error pages with custom error pages?

Reply:
> Do nginx' standard error pages contain nginx version information?

It depends on the `server_tokens` directive[1], earlier it was set to `yes`:
```
<html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>openresty/1.13.6.1</center>
</body>
</html>
```
with `server_tokens` set to `off`:
```
<html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>openresty</center>
</body>
</html>
```

My goal here was to:
a) prevent Nginx version fingerprinting basing on the structure of the error page itself. Both now and in the future. The version of the Nginx disclosed here does not really matter as the same information can be obtained by looking at the `Server` header.
b) hit two birds with one stone by enabling custom error pages implementation by the UI team as per [2]

The paragraph a) may seem as a bit of an overkill, but I was afraid that if we do not address all the obvious ways to fingerprint our software, then our customer's auditors will come back to us sooner or later.

[1] http://nginx.org/en/docs/http/ngx_http_core_module.html#server_tokens
[2] https://jira.mesosphere.com/browse/DCOS_OSS-1938

> Does this list here replace all nginx standard error pages with custom error pages?

No, it does not. But this was not my intention - I just wanted something that would cover majority of the cases. Would you like me to be more thorough here?